test/logflare/context_cache_test.exs

```diff
@@ -0,0 +1,38 @@
+defmodule Logflare.ContextCacheTest do
+  use Logflare.DataCase, async: false
+
+  import Logflare.Factory
+
+  alias Logflare.ContextCache
+  alias Logflare.Sources
+
+  setup do
+    user = insert(:user)
+    insert(:plan, name: "Free")
+    source = insert(:source, user: user)
+    %{source: source}
+  end
+
+  test "ContextCache works", %{source: source} do
+    context = Sources
+    fun_arity = {:get_by, 1}
+    args = [[token: source.token]]
+
+    # Cache our function call results
+    cached_source = ContextCache.apply_fun(context, fun_arity, args)
```

Comment:
apply_fun could probably be simplfiied to receive `apply_fun({Sources, :get_by, [[token: source.token]]} )`, since the arity is implied through the arg count.

Reply:
or even `apply_fun(Sources, :get_by, [...])`

Reply:
Pull the arity out of the cache key but you kind of want it here because it makes calling apply_fun easier.